framework/src/outputs/CSV.C

```diff
@@ -172,28 +178,46 @@ CSV::output(const ExecFlagType & type)
         it.second.sortColumns();
 
       auto include_time_suffix = !vpp_data.containsCompleteHistory(vpp_name);
-      std::string fname = getVectorPostprocessorFileName(vpp_name, include_time_suffix);
-      std::string fprefix = getVectorPostprocessorFilePrefix(vpp_name);
-      _latest_vpp_filenames.emplace_back(fname, fprefix);
-      it.second.printCSV(fname, 1, _align);
+      auto is_distributed = vpp_data.isDistributed(vpp_name);
 
-      if (_create_latest_symlink)
+      if (is_distributed || processor_id() == 0)
       {
-        std::string out_latest = fprefix + "_LATEST.csv";
-        MooseUtils::createSymlink(fname, out_latest);
-      }
+        std::string fname =
+            getVectorPostprocessorFileName(vpp_name, include_time_suffix, is_distributed);
+        std::string fprefix = getVectorPostprocessorFilePrefix(vpp_name);
+
+        _latest_vpp_filenames.emplace_back(fname, fprefix, is_distributed);
+
+        it.second.printCSV(fname, 1, _align);
 
-      if (_time_data)
-        _vector_postprocessor_time_tables[vpp_name].printCSV(fprefix + "_time.csv");
+        if (_create_latest_symlink)
+        {
+          std::ostringstream out_latest;
+          out_latest << fprefix << "_LATEST.csv";
+          if (is_distributed)
+            out_latest << "." << processor_id();
```

Comment:
I would recommend that you format the `processor_id` now before we start checking in a bunch of tests (e.g. give it a reasonable fixed width, 001, 002, 003, etc). That makes sorting and processing a little bit easier.